bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -251,7 +255,7 @@ public Rule add(Rule rule) {
      * @param rule candidate for unique ID
      * @return a rule with UID
      */
-    protected Rule initRuleId(String rUID, Rule rule) {
+    protected @Nonnull Rule initRuleId(String rUID, Rule rule) {
```

Comment:
No, that is the whole intention of the annotations (please read all the discussion on https://github.com/eclipse/smarthome/issues/3624): Nobody reads JavaDoc and if you can express the very same in a machine readable way, it is WAY BETTER.